.github/workflows/build-plugin-zip.yml

```diff
@@ -56,7 +56,7 @@ jobs:
                 github.event.inputs.version == 'rc' ||
                 github.event.inputs.version == 'stable'
               ) || (
-                endsWith( github.ref, needs.compute-stable-branches.outputs.current_stable_branch ) &&
+                startsWith( github.ref, 'refs/heads/release/' ) &&
```

Comment:
any reason for this change?

Reply:
Yes; the whole condition looked like this:

```
            github.event_name == 'workflow_dispatch' && (
              (
                github.ref == 'refs/heads/trunk' ||
                endsWith( github.ref, needs.compute-stable-branches.outputs.next_stable_branch )
              ) && (
                github.event.inputs.version == 'rc' ||
                github.event.inputs.version == 'stable'
              ) || (
                endsWith( github.ref, needs.compute-stable-branches.outputs.current_stable_branch ) &&
                github.event.inputs.version == 'stable'
              )
            )
```

The first part until the last `||` binary op, deals with current or next releases:
* If the branch is `trunk` and the input is `rc/stable` it will evaluate to `true`. That will proceed to other steps that will check the input again to decide what to do next (i.e create a new branch if RC, or switch to it if stable. 

_I'm not entirely sure about the `endsWith( github.ref, needs.compute-stable-branches.outputs.next_stable_branch )` expression, but I think it's there to allow the user to trigger the workflow either from `trunk` or from the latest release branch, i.e make them equivalent (so, let's say that the last release is 15.6), triggering the workflow from `trunk` or `15.6` would yield the same result, because it's the tip of releases._

As you might now, the main use cases for releases are, which are documented [here](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/release.md), are:
* `trunk` + `stable` will result in a new non-RC stable release if the version logged in the branch's `package.json` is an RC.
* `trunk` + `stable` will result in a patch release if the version in the branch's `package.json` is stable.
* `trunk` + `rc` will result in a new minor/major (it depends) and will also create a new release branch.

This covers the conditions until the last OR operator. Now onto this expression:

```
(
    endsWith( github.ref, needs.compute-stable-branches.outputs.current_stable_branch ) &&
    github.event.inputs.version == 'stable'
)
```

This deals with the cases when the releaser selected a branch other than `trunk` _or_  the last release branch (the tip of the list of versions). This was there to allow point releases for the current stable release. As long as the next release was an RC, this expression would return true as the `current_stable_branch` would be the same as the current ref; but if there was a more recent stable release, and if you were releasing for a previous version branch, then `current_stable_branch` would point to the more recent stable, and this would evaluate to `false`. The `bump-version` step would be skipped, and with it, pretty much the whole workflow would fail, as all the subsequent tasks depend on it.

That leads us to the change in this PR:

```
(
    startsWith( github.ref, 'refs/heads/release/' ) &&
    github.event.inputs.version == 'stable'
)
```

This was done to allow this workflow to be triggered for any previous release branches. This means that the user can now run this for any past branch and not only for the `current_stable_branch`. Of course, this wasn't the only change needed for it to work, but it was the first one that allowed this first workflow to be triggered for any past branches.

Reply:
And yes, it's a bit convoluted. I had to re-read this many times before writing the explanation above. Let me know if you have any other specific questions as I might have missed something, too!

Reply:
Bit late to this, but your explanation is spot-on, @fullofcaffeine. Makes me wonder if we should add some comments to the expression (or those would just make it harder to parse 😅 )

> _I'm not entirely sure about the `endsWith( github.ref, needs.compute-stable-branches.outputs.next_stable_branch )` expression, but I think it's there to allow the user to trigger the workflow either from `trunk` or from the latest release branch, i.e make them equivalent (so, let's say that the last release is 15.6), triggering the workflow from `trunk` or `15.6` would yield the same result, because it's the tip of releases._

That's exactly right! My guiding principle for these workflows was to be as forgiving as possible to the user, as long as their intention could be ascertained. In this case, the intention is clear enough: they want to trigger a stable release for the current branch, and they shouldn't have to worry whether to select the `release/` branch (as they're familiar with from publishing point releases for previous release branches), or `trunk` -- it should "just work" either way.